internal/rootcoord/meta_table.go

```diff
@@ -1456,3 +1461,118 @@ func (mt *MetaTable) RestoreRBAC(ctx context.Context, tenant string, meta *milvu
 
 	return mt.catalog.RestoreRBAC(mt.ctx, tenant, meta)
 }
+
+// check if the privielge group name is defined by users
+func (mt *MetaTable) IsCustomPrivilegeGroup(groupName string) (bool, error) {
+	privGroups, err := mt.catalog.ListPrivilegeGroups(mt.ctx)
+	if err != nil {
+		return false, err
+	}
+	for _, group := range privGroups {
+		if group.GroupName == groupName {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (mt *MetaTable) CreatePrivilegeGroup(groupName string) error {
+	if funcutil.IsEmptyString(groupName) {
+		return fmt.Errorf("the privilege group name is empty")
+	}
+	mt.permissionLock.Lock()
+	defer mt.permissionLock.Unlock()
+
+	definedByUsers, err := mt.IsCustomPrivilegeGroup(groupName)
+	if err != nil {
+		return err
+	}
+	if definedByUsers {
+		return merr.WrapErrParameterInvalidMsg("privilege group name [%s] is defined by users", groupName)
+	}
+	if util.IsPrivilegeNameDefined(groupName) {
+		return merr.WrapErrParameterInvalidMsg("privilege group name [%s] is defined by built in privileges or privilege groups in system", groupName)
+	}
+	privileges := make([]*milvuspb.PrivilegeEntity, 0)
+	return mt.catalog.AlterPrivilegeGroup(mt.ctx, groupName, privileges)
+}
+
+func (mt *MetaTable) DropPrivilegeGroup(groupName string) error {
+	if funcutil.IsEmptyString(groupName) {
+		return fmt.Errorf("the privilege group name is empty")
+	}
+	mt.permissionLock.Lock()
+	defer mt.permissionLock.Unlock()
+
+	definedByUsers, err := mt.IsCustomPrivilegeGroup(groupName)
+	if err != nil {
+		return err
+	}
+	if !definedByUsers {
+		return merr.WrapErrParameterInvalidMsg("there is no privilege group name [%s] to drop", groupName)
+	}
+	if util.IsPrivilegeNameDefined(groupName) {
```

Comment:
unnecessary check, if it's a defined privilege group, it can't be a built in privilege